Networking/NetworkingTests/Mapper/PaymentGatewayMapperTests.swift

```diff
@@ -0,0 +1,62 @@
+import XCTest
+@testable import Networking
+
+final class PaymentGatewayMapperTests: XCTestCase {
+
+    /// Dummy Site ID.
+    ///
+    fileprivate let dummySiteID: Int64 = 12983476
+
+    /// Verifies that the PaymentGateway is parsed.
+    ///
+    func test_PaymentGateway_map_parses_all_paymentGateways_in_response() throws {
+        let paymentGateway = try mapRetrievePaymentGatewayResponse()
+        XCTAssertNotNil(paymentGateway)
+    }
+
+    /// Verifies that the `siteID` is added in the mapper, because it's not provided by the API endpoint
+    ///
+    func test_PaymentGatewaysList_map_includes_siteID_in_parsed_results() throws {
+        let paymentGateway = try mapRetrievePaymentGatewayResponse()
+        XCTAssertEqual(paymentGateway.siteID, dummySiteID)
+    }
+
+    /// Verifies that the fields are all parsed correctly
+    ///
+    func test_PaymentGatewaysList_map_parses_all_fields_in_result() throws {
+        let paymentGateway = try mapRetrievePaymentGatewayResponse()
+
+        let expectedPaymentGateway = PaymentGateway(siteID: dummySiteID,
+                                                    gatewayID: "cod",
+                                                    title: "Cash on delivery",
+                                                    description: "Pay with cash upon delivery.",
+                                                    enabled: true,
+                                                    features: [.products])
+
+        XCTAssertEqual(paymentGateway, expectedPaymentGateway)
+    }
+}
+
+
+// MARK: - Test Helpers
+///
+private extension PaymentGatewayMapperTests {
+
+    /// Returns the PaymentGatewayMapper output upon receiving `filename` (Data Encoded)
+    ///
+    func mapPaymentGateway(from filename: String) throws -> PaymentGateway {
+        guard let response = Loader.contentsOf(filename) else {
+            throw FileNotFoundError()
+        }
+
+        return try PaymentGatewayMapper(siteID: dummySiteID).map(response: response)
+    }
+
+    /// Returns the PaymentGatewayMapper output from `paymentGateway.json`
```

Comment:
@iamgabrielma Sort of... but no. Your description of a stub is bang on, if you remove the word "response": 

> A tiny replacement for the network ~response~ so it always returns some fixed JSON, and we do not have to hit the Network.

A stub is a test double, where "double" has the same meaning as in "stunt double" 😅. 

We use test doubles in place of dependencies, i.e. other code which the code under test needs to do its job.

A simple `Mapper` like this does not have any dependencies, because its job is to implement

```
func map(response: Data) throws -> Output
```

For a particular `Output` type. Notice that it gets given the `response` data, it doesn't have to get it from the network or anywhere else: that's the Remote's job. A mapper can be implemented as a function without dependencies: it takes the data as input, and tries to parse it, and returns the object in question.

This is therefore a helper method, which passes a known piece of test data (`payment-gateway-cod.json`) to the system under test (`PaymentGatewayMapper`).

If you wanted to test this same functionality at the [`PaymentGatewayRemote` level](https://github.com/woocommerce/woocommerce-ios/blob/trunk/Networking/Networking/Remote/PaymentGatewayRemote.swift), you would have dependencies, in particular, `Network`. All our remotes are subclasses of `Remote`, which has a `network` property. Generally, the remotes call `enqueue`, to send a request to the network and expect the relevant response, which they pass through the appropriate mapper.

You could pass a `StubPaymentGatewayNetwork` [instance to `PaymentGatewayRemote(network:)` as a test double](https://github.com/woocommerce/woocommerce-ios/blob/010953a692dc67a0d8d0412f3eef500ba2699566/Networking/Networking/Remote/Remote.swift#L20). Then the [`enqueue` call](https://github.com/woocommerce/woocommerce-ios/blob/010953a692dc67a0d8d0412f3eef500ba2699566/Networking/Networking/Remote/PaymentGatewayRemote.swift#L16) would make a [`network.responseData` request which would go to your stub](https://github.com/woocommerce/woocommerce-ios/blob/010953a692dc67a0d8d0412f3eef500ba2699566/Networking/Networking/Remote/Remote.swift#L32), and the stub would return the test data without going to the real network.

If you did that, it would have to implement everything on the [`Network` protocol](https://github.com/woocommerce/woocommerce-ios/blob/010953a692dc67a0d8d0412f3eef500ba2699566/Networking/Networking/Network/Network.swift#L20), but in a very basic way, e.g. just empty implementations for any methods you're not calling, and always returning the test data in the `responseData(for request:)` implementations.

Reply:
> Not a blocker, but should we change the comment from `paymentGateway.json` to `payment-gateway-cod.json`?

Yep, I'll do that. Good spot.

Reply:
Gotcha! Thanks for the explanation! 🙇